packages/eslint-config/index.js

```diff
@@ -0,0 +1,132 @@
+// =============================================================================
+
+module.exports = {
+  env: {
+    browser: true,
+  },
+  extends: ["react-app", "airbnb", "plugin:prettier/recommended"],
+  ignorePatterns: ["node_modules/", "build/", "public/"],
+  plugins: ["simple-import-sort"],
+  rules: {
+    "func-names": 0,
+    // don't require extensions for typescript modules
+    "import/extensions": [
+      "error",
+      "always",
+      { js: "never", ts: "never", tsx: "never" },
+    ],
+    "import/no-extraneous-dependencies": ["error", { packageDir: "./" }],
+    "import/prefer-default-export": "off",
+    "jsx-a11y/label-has-associated-control": [
+      "error",
+      {
+        required: {
+          some: ["nesting", "id"],
+        },
+      },
+    ],
+    "jsx-a11y/label-has-for": [
+      "error",
+      {
+        required: {
+          some: ["nesting", "id"],
+        },
+      },
+    ],
+    "no-new": 1,
+    "no-restricted-imports": [
+      "error",
+      {
+        paths: [
+          {
+            name: "styled-components",
+            message: "Please import from styled-components/macro.",
+          },
+          {
+            name: "react-spring",
+            message:
+              "For IE 11 support, please import from react-spring/web.cjs",
+          },
+          {
+            name: "react-spring/web",
+            message:
+              "For IE 11 support, please import from react-spring/web.cjs",
+          },
+          {
+            name: "react-spring/renderprops",
+            message:
+              "For IE 11 support, please import from react-spring/renderprops.cjs",
+          },
+        ],
+      },
+    ],
+
+    // we are only targeting es5 environments so we don't have to pass 10
+    radix: ["error", "as-needed"],
+    // conflicts with prettier
+    "react/jsx-curly-newline": 0,
+    "react/jsx-filename-extension": ["error", { extensions: [".tsx", ".js"] }],
+    // conflicts with prettier
+    "react/jsx-indent": "off",
+    // conflicts with prettier
+    "react/jsx-one-expression-per-line": 0,
+    "react/jsx-props-no-spreading": 0,
+    "react/jsx-wrap-multilines": [
+      "error",
+      { declaration: false, assignment: false },
+    ],
+  },
+  overrides: [
+    {
+      extends: ["plugin:@typescript-eslint/recommended"],
+      files: ["**.ts", "**.tsx"],
+      parser: "@typescript-eslint/parser",
+      rules: {
+        // with static typing this rule is not so useful
+        "consistent-return": "off",
+        // these bare ESLint rules are superseded by TS equivalents
+        "no-shadow": "off",
+        "@typescript-eslint/no-shadow": "error",
+        "no-use-before-define": "off",
+        "@typescript-eslint/no-use-before-define": ["error"],
+        // TypeScript makes these redundant
+        "react/prop-types": "off",
+        "react/require-default-props": "off",
+      },
+    },
+    {
+      files: ["**.test.js"],
```

Comment:
For the overly eager, should we add TS here as well?
```
files: ["**.test.js", "**.test.ts"],
```

Reply:
hmm good callout, I think probably you are right. let me look into that quickly before merging